aca_preview/preview.py

```diff
@@ -0,0 +1,329 @@
+        aca.outdir = outdir
+        aca.context = {}
+        aca.messages = []
+        aca.preview()
+        acas.append(aca)
+
+    context = {}
+
+    context['load_name'] = load_name.upper()
+    context['version'] = VERSION
+    context['acas'] = acas
+    context['summary_text'] = get_summary_text(acas)
+
+    template_file = FILEDIR / 'index_template_preview.html'
+    template = Template(open(template_file, 'r').read())
+    out_html = template.render(context)
+
+    out_filename = outdir / 'index.html'
+    with open(out_filename, 'w') as fh:
+        fh.write(out_html)
+
+
+def stylize(text, category):
+    """Stylize ``text``.
+
+    Currently ``category`` of critical, warning, caution, or info are supported
+    in the CSS span style.
+
+    """
+    out = f'<span class="{category}">{text}</span>'
+    return out
+
+
+def get_acas(load_name):
+    filenames = [f'{load_name}_proseco.pkl', f'{load_name}.pkl']
+    for filename in filenames:
+        if Path(filename).exists():
+            acas = pickle.load(open(filename, 'rb'))
+            return acas
+    raise FileNotFoundError(f'no matching pickle file {filenames}')
+
+
+def get_summary_text(acas):
+    """Get summary text for all catalogs.
+
+    This is like::
+
+      Proseco version: 4.4-r528-e9d6c73
+
+      OBSID = -3898   at 2019:027:21:58:37.828   7.8 ACQ | 5.0 GUI |
+      OBSID = 21899   at 2019:028:01:17:39.066   7.8 ACQ | 5.0 GUI |
+
+    """
+    lines = []
+    for aca in acas:
+        line = (f'<a href="#obsid{aca.obsid}">OBSID = {aca.obsid:7s}</a>'
+                f' at {aca.date}   '
+                f'{aca.acq_count:.1f} ACQ | {aca.guide_count:.1f} GUI |')
+
+        # Warnings
+        for category in CATEGORIES:
+            msgs = [msg for msg in aca.messages if msg['category'] == category]
+            if msgs:
+                text = stylize(f' {category.capitalize()}: {len(msgs)}', category)
+                line += text
+
+        lines.append(line)
+
+    return '\n'.join(lines)
+
+
+class ACAReviewTable(ACATable):
+    def make_starcat_plot(self):
+        plotname = f'cat{self.obsid}.png'
+        outfile = self.outdir / plotname
+        self.context['catalog_plot'] = plotname
+
+        if outfile.exists():
+            return
+
+        stars = StarsTable.from_agasc(self.att, date=self.date)
+        self.stars = stars
+
+        fig = plt.figure(figsize=(4.5, 4))
+        ax = fig.add_subplot(1, 1, 1)
+        self.acqs.stars
+        self.plot(ax=ax)
+        plt.tight_layout()
+        fig.savefig(str(outfile))
+        plt.close(fig)
+
+    def get_text_pre(self):
+        """Get pre-formatted text for report."""
+
+        P2 = -np.log10(self.acqs.calc_p_safe())
+        att = Quat(self.att)
+        self._base_repr_()
+        catalog = '\n'.join(self.pformat(max_width=-1))
+        self.acq_count = np.sum(self.acqs['p_acq'])
+        self.guide_count = self.guides.guide_count(self.guides['mag'], self.guides.t_ccd)
```

Comment:
Sorry, I had included that `guide_count` method in the PR because I wanted the "new" fractional count.  So this should go back to being from chandra_aca.star_probs, but with the PR I put in .  Sigh.

Reply:
Gotcha.  I figured as much but wasn't sure if there was a hidden proseco dependence in aca_preview on a newer version etc.  Good to know.

Reply:
Though actually, does it even matter if it is the old count or the new fractional count at this point?  New fractional count preferred but I don't see any checks on it here yet anyway.